app/models/journeyDomain/locationOfGoods/LocationOfGoodsDomain.scala

```diff
@@ -50,24 +42,30 @@ object LocationOfGoodsDomain {
   implicit val userAnswersReader: UserAnswersReader[LocationOfGoodsDomain] =
     LocationTypePage.reader.flatMap {
       typeOfLocation =>
-        val identifierReads = InferredIdentificationPage.reader orElse IdentificationPage.reader
-        identifierReads.flatMap {
-          case CustomsOfficeIdentifier => LocationOfGoodsV.userAnswersReader(typeOfLocation)
-          case EoriNumber              => LocationOfGoodsX.userAnswersReader(typeOfLocation)
-          case AuthorisationNumber     => LocationOfGoodsY.userAnswersReader(typeOfLocation)
-          case UnlocodeIdentifier      => LocationOfGoodsU.userAnswersReader(typeOfLocation)
-          case CoordinatesIdentifier   => LocationOfGoodsW.userAnswersReader(typeOfLocation)
-          case AddressIdentifier       => LocationOfGoodsZ.userAnswersReader(typeOfLocation)
-          case PostalCode              => LocationOfGoodsT.userAnswersReader(typeOfLocation)
+        typeOfLocation match {
+          case LocationType("B", _) => LocationOfGoodsY.userAnswersReader()
```

Comment:
This line won't be needed once we reintroduce the change in `IdentificationController`